crates/sui-core/src/checkpoints/checkpoint_executor/mod.rs

```diff
@@ -503,7 +509,86 @@ pub async fn execute_checkpoint(
     Ok(())
 }
 
-fn extract_tx_data(
+fn assert_not_forked(
+    checkpoint: &VerifiedCheckpoint,
+    tx_digest: &TransactionDigest,
+    expected_digest: &TransactionEffectsDigest,
+    actual_effects: &TransactionEffects,
+) {
+    if *expected_digest != actual_effects.digest() {
+        // log observed effects (too big for panic message) and then panic.
+        error!(
+            ?checkpoint,
+            ?tx_digest,
+            ?expected_digest,
+            ?actual_effects,
+            "fork detected!"
+        );
+        panic!(
+            "When executing checkpoint {}, transaction {} \
+            is expected to have effects digest {}, but got {}!",
+            checkpoint.sequence_number(),
+            tx_digest,
+            expected_digest,
+            actual_effects.digest(),
+        );
```

Comment:
Why both log lines? Why not just the panic?

Reply:
i want to print the full effects in the log, which seems like a lot for a panic. also, panic messages are usually much harder to find with loki.